### PR TITLE
Manage classAttended2019 level value in pre populate student informat…

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**
…ion.

### CODE REVIEW

File: src/components/AdvanceSearch.jsx [hunk 1]
```diff
@@ -18,16 +18,16 @@ class AdvanceSearch extends Component {
     this.onChangeMultipleIdSearchCheckBox = this.onChangeMultipleIdSearchCheckBox.bind(this);
   }
 
-  setInputValue(e){
-    if(isEmpty(e.target.value)){
+  setInputValue(e) {
```
Comment (mohitsaladiya) [inline src/components/AdvanceSearch.jsx]: Add JSDoc.

File: src/components/AdvanceSearch.jsx [hunk 2]
```diff
@@ -86,44 +84,54 @@ class AdvanceSearch extends Component {
...
-      let searchResult = [];
-      for(let index in searchStudentsIds){
-        let result = this.props.students.filter(obj => {
-          return obj.id === Number(searchStudentsIds[index]);
-        });
+    } else {
+      const searchStudentsIds = this.state.inputValue.split(',');
+      const searchResult = [];
+      for (const index in searchStudentsIds) {
+        const result = this.props.students.filter(obj => obj.id === Number(searchStudentsIds[index]));
```
Comment (mohitsaladiya) [inline src/components/AdvanceSearch.jsx]:
```suggestion:
        const result = this.props.students.filter(student => student.id === Number(searchStudentsIds[index]));
```

File: src/components/AdvanceSearch.jsx [hunk 3]
```diff
@@ -86,44 +84,54 @@ class AdvanceSearch extends Component {
...
-            <input type="text" onChange={this.setInputValue} value={this.state.inputValue} className = "search-input-advance" />
+        <div className="input-radio">
+          <label htmlFor="search_input" className="input-text">
+            <input type="text" onChange={this.setInputValue} value={this.state.inputValue} className="search-input-advance" />
             <button type="submit" form="advanceSearch" value="Submit" className="search" onClick={this.advanceSearch}>
-              <i className="fa fa-search"/>
+              <i className="fa fa-search" />
             </button>
           </label>
-          {/*<button type="reset" value="Reset" onClick={this.clearFilter} className = "advance-search-button display-none">
```
Comment (mohitsaladiya) [inline src/components/AdvanceSearch.jsx]: Remove commented code if not required. OR
Mention it's purpose for being commented.

File: src/components/AdvanceSearch.jsx [hunk 4]
```diff
@@ -86,44 +84,54 @@ class AdvanceSearch extends Component {
...
+          {/* <button type="reset" value="Reset" onClick={this.clearFilter} className = "advance-search-button display-none">
             <i className="fa fa-trash card-icon"/>Clear
           </button>*/}
-          <div className = "advance-input-radio">
-           {/* <div className="input-radio-container display-none">
+          <div className="advance-input-radio">
+            {/* <div className="input-radio-container display-none">
               <input type="checkbox" name="thresholdValue" value="0.0" onClick={this.onClickRadioButton}  defaultChecked />
```
Comment (mohitsaladiya) [inline src/components/AdvanceSearch.jsx]: Here as well.

File: src/components/StudentRegistrationCorrectionForm.jsx [hunk 5]
```diff
@@ -33,10 +33,12 @@ import {
   isFetched,
   isLoading,
   isUpdated,
+  getUserId,
+  getUserSecretKey,
 } from '../reducers/studentRegistrationReducer';
 import SelectListInputField from './formComponents/SelectListInputField';
 import Button from './commonComponents/Button';
-import { getUserId, getUserSecretKey } from '../reducers/studentRegistrationReducer';
+import isEmpty from 'lodash/isEmpty';
```
Comment (mohitsaladiya) [inline src/components/StudentRegistrationCorrectionForm.jsx]: Third-party module imports should always above project-level imports. Add a blank line between third-party imports and project-level imports.

File: src/components/StudentRegistrationCorrectionForm.jsx [hunk 6]
```diff
@@ -89,12 +91,33 @@ class StudentRegistrationCorrectionForm extends Component {
...
+   * @param {object} nextProps
+ */
   prePopulateCourse2019(nextProps) {
-    const lastCourse = nextProps.studentData.classAttended2017;
+    const lastCourse = nextProps.studentData.classAttended2018;
     const level = checkLevelValue(lastCourse);
+    // In classAttended2018 Level is greater than 0 (level > 0) condition will satisfied.
     if (level > 0) {
-      const updatedData = extend(cloneDeep(this.state.student), { classAttended2019: level + 1 });
+      // In classAttended2018 Level is greater than 7 like 'Level 8' in that condition will pre populate
```
Comment (mohitsaladiya) [inline src/components/StudentRegistrationCorrectionForm.jsx]: This logic should be in a util file.

File: src/components/StudentRegistrationCorrectionFormURL.jsx [hunk 7]
```diff
@@ -99,12 +99,33 @@ class StudentRegistrationCorrectionFormURL extends Component {
...
     });
   }
-
+  /**
+   * prePopulateCourse2019 method will use for pre populate the information of fetch student.
+   * @param {object} nextProps
+   */
   prePopulateCourse2019(nextProps) {
-    const lastCourse = nextProps.studentData.classAttended2017;
+    const lastCourse = nextProps.studentData.classAttended2018;
```
Comment (mohitsaladiya) [inline src/components/StudentRegistrationCorrectionFormURL.jsx]: use util method for this purpose.

File: src/components/AdvanceSearch.jsx (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/components/AdvanceSearch.jsx]: Done.

File: src/components/AdvanceSearch.jsx (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/components/AdvanceSearch.jsx]: Done.

File: src/components/AdvanceSearch.jsx (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/components/AdvanceSearch.jsx]: Done.

File: src/components/AdvanceSearch.jsx (hunk 4 above)
Comment (BhagyashreeBhoyar) [reply src/components/AdvanceSearch.jsx]: Done.

File: src/components/StudentRegistrationCorrectionForm.jsx (hunk 5 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentRegistrationCorrectionForm.jsx]: Done.

File: src/components/StudentRegistrationCorrectionForm.jsx (hunk 6 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentRegistrationCorrectionForm.jsx]: Done.

File: src/components/StudentRegistrationCorrectionFormURL.jsx (hunk 7 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentRegistrationCorrectionFormURL.jsx]: Done.

File: src/utils/registrationFormUtils.js [hunk 8]
```diff
@@ -214,32 +215,58 @@ export const requireFieldsValidate = (value, name) => {
...
+  const level = value ? value.slice(6, 8) : -1;
+  return (Number(level));
+};
+/**
+ * updateStudentDataAccordingClassAttended2018Level method manipulate the student data
+ * according to classAttended level value of previous year.
+ * @param {Object} studentData
+ * @return {Object} studentData
+ */
+export const updateStudentDataAccordingClassAttended2018Level = (studentData) => {
```
Comment (mohitsaladiya) [inline src/utils/registrationFormUtils.js]: Rename it to updateClassAttended2019InStudentData

**Pull request merged**